packages/admin/cms-admin/src/blocks/createTextLinkBlock.tsx

```diff
@@ -40,6 +40,20 @@ export function createTextLinkBlock({ link: LinkBlock, name = "TextLink" }: Crea
 
         definesOwnPadding: true,
 
+        extractTextContents: (state) => {
+            return state.text.trim().length > 0 ? [state.text] : [];
+        },
+
+        replaceTextContents: (state, contents) => {
+            const translation = contents.find((content) => content.original === state.text);
+            const text = translation && translation.replaceWith !== "" ? translation.replaceWith : state.text;
+
+            return {
+                ...state,
+                text,
+            };
+        },
+
```

Comment:
This doesn't extract the contents of its child link block... could there be a case where the child link block may define additional texts?